addon/components/listing.js

```diff
@@ -63,7 +83,7 @@ export default class ListingComponent extends Component {
     //  'You attempted to update `subForms` on `ListingComponent`, but it had already been used previously in the same computation. '
     // Why is it triggered? this.subForms is tracked, is rendered before the method below (which has side effects, i.e. updates this.subForms) ran.
     next(this, () => {
-      this.renderSubForms();
+      this.renderSubForms({ orderSubForms: true });
```

Comment:
We only do the sorting when the data is initially loaded, after that the ordering can't get out of order, since we only add new forms after the already existing ones.